Makefile.in

```diff
@@ -527,3 +527,42 @@ preexist:
 
 maintainer-dist: preexist tag
 	svnorcvs release $(DISTTAGNAME)
+
+######################################################################
+# Generated code snapshot and diff for tests
+
+# Can be overridden for multiple snapshots
+TEST_SNAP_DIR ?= test_regress/snapshot
+
+# Command to diff directories
+TEST_DIFF_TOOL ?= $(if $(shell which icdiff), icdiff -N -r, diff -r)
+
+TEST_SNAP_IGNORE := *.status *.log *.dat *.d *.o *.a *.so *.txt *.html \
+                    *.includecache *.out *.fst *.fst.vcd  *.tree *.dot \
+                    *.csv *.xml *.hash *.cmake gmon.out.* CMakeFiles \
```

Comment:
The idea with this was that the diff should be completely empty in an ideal world, though there is still some nondeterminism.

I changed .txt to stats.txt, (stats contains timing info, which varies).

.out should not change but the test should fail if it did.

.log contains all kinds of noise like timestamps and memory addresses, and the tests that generally care about should fail

.vcd is only included because we are no longer printing the $date header, so is stable, and vcddiff does not pick up event ordering changes. Strangely there is one unstable vcd with no code changes.